Modules/posixmodule.c

```diff
@@ -10035,6 +10040,205 @@ os_times_impl(PyObject *module)
 #endif /* HAVE_TIMES */
 
 
+#if defined(HAVE_TIMERFD_API)
+#define ONE_SECOND_IN_NS (1000 * 1000 * 1000)
+#define EXTRACT_NSEC(value)  (long)( ( (double)(value) - (time_t)(value) ) * 1e9)
+
+static PyObject *
+build_itimerspec(const struct itimerspec* curr_value)
+{
+    double _interval = (double)(curr_value->it_interval.tv_sec) +
+                       (double)curr_value->it_interval.tv_nsec * 1e-9;
+    PyObject *interval = PyFloat_FromDouble(_interval);
+    if (interval == NULL) {
+        return NULL;
+    }
+    double _value = (double)(curr_value->it_value.tv_sec) +
+                    (double)curr_value->it_value.tv_nsec * 1e-9;
+    PyObject *value = PyFloat_FromDouble(_value);
+    if (value == NULL) {
+        Py_DECREF(interval);
+        return NULL;
+    }
+    PyObject *tuple = PyTuple_Pack(2, interval, value);
+    Py_DECREF(interval);
+    Py_DECREF(value);
+    return tuple;
+}
+
+static PyObject *
+build_itimerspec_ns(const struct itimerspec* curr_value)
+{
+    return PyTuple_Pack(
+        2,
+        PyLong_FromLongLong((long long)curr_value->it_interval.tv_sec * ONE_SECOND_IN_NS + (long long)curr_value->it_interval.tv_nsec),
+        PyLong_FromLongLong((long long)curr_value->it_value.tv_sec * ONE_SECOND_IN_NS + (long long)curr_value->it_value.tv_nsec)
+    );
+}
+
+/*[clinic input]
+os.timerfd_create
+
+    clockid: int
+        CLOCK_REALTIME or CLOCK_MONOTONIC. CLOCK_REALTIME.
+    flags: int
+        0 or  a bit mask of TFD_NONBLOCK or TFD_CLOEXEC.
```

Comment:
I think *flags* should default to 0, and perhaps should also be keyword-only. At the least, it should be positional-or-keyword.

Reply:
@AA-Turner 
Changed to keyword only except for fd.